### PR TITLE
@orta: Adds Optional isEmpty-ness.

### PR DESCRIPTION
**Pull request by ashfurrow**
Also upgraded Kiosk project to Xcode's recommended settings.

Let me know what you think!

### CODE REVIEW

File: Kiosk.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1323,6 +1323,7 @@
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
```
Comment (orta) [inline Kiosk.xcodeproj/project.pbxproj]: :star2:

File: KioskTests/App/SwiftExtensionsTests.swift [hunk 2]
```diff
@@ -1,8 +1,88 @@
...
-import Kiosk
 import ReactiveCocoa
 
+@testable
+import Kiosk
+
+let Empty = "empty"
+let NonEmpty = "nonEmpty"
+
+// Wordaround due to limitations in the compiler. See: https://twitter.com/jckarter/status/636969467713458176
```
Comment (orta) [inline KioskTests/App/SwiftExtensionsTests.swift]: Wordaround

File: KioskTests/App/SwiftExtensionsTests.swift [hunk 3]
```diff
@@ -21,5 +101,18 @@ class SwiftExtensionsTests: QuickSpec {
...
+            itBehavesLike("an Occupiable") { [ Empty: AnyOccupiable(value: ""), NonEmpty: AnyOccupiable(value: "hi") ] }
+        }
+
+        describe("Array") {
+            itBehavesLike("an Occupiable") { [ Empty: AnyOccupiable(value: Array<String>()), NonEmpty: AnyOccupiable(value: ["hi"]) ] }
+        }
+
+        describe("Dictionary") {
+            itBehavesLike("an Occupiable") { [ Empty: AnyOccupiable(value: Dictionary<String, String>()), NonEmpty: AnyOccupiable(value: ["hi": "yo"]) ] }
+        }
```
Comment (orta) [inline KioskTests/App/SwiftExtensionsTests.swift]: great use of the examples thing in tests

File: Kiosk/App/SwiftExtensions.swift [hunk 4]
```diff
@@ -17,8 +17,40 @@ public extension String {
...
         return !isEmpty
     }
 }
+
+extension String: Occupiable { }
+
+// I can't think of a way to combine these collection types. Suggestions welcome.
+extension Array: Occupiable { }
+extension Dictionary: Occupiable { }
+extension Set: Occupiable { }
```
Comment (paulyoung) [inline Kiosk/App/SwiftExtensions.swift]: Can you extend `CollectionType` instead?

Comment (ashfurrow) [inline Kiosk/App/SwiftExtensions.swift]: I tried that but it didn't work – any suggestions?

Comment (ole) [inline Kiosk/App/SwiftExtensions.swift]: It seems extending protocols to conform to other protocols isn't implemented yet. https://twitter.com/jckarter/status/616254698509611008

Comment (ashfurrow) [inline Kiosk/App/SwiftExtensions.swift]: Ah, that explains it then. Thanks @ole!

Comment (paulyoung) [inline Kiosk/App/SwiftExtensions.swift]: Sorry for the confusion. I [extended `CollectionType` in Ogra](https://github.com/edwardaux/Ogra/pull/1/files) but we're using Swift 2.

Comment (paulyoung) [inline Kiosk/App/SwiftExtensions.swift]: Ahh - what I'm doing there is different anyway.

**Pull request merged**